instances/widgets.treasury-factory.near/widget/pages/settings/MembersEditor.jsx

```diff
@@ -28,6 +31,32 @@ const [showDeleteModal, setShowDeleteModal] = useState(false);
 const [showCancelModal, setShowCancelModal] = useState(false);
 const [memberAlreadyExistRoles, setMemberAlreadyExistRoles] = useState(null);
 const [showErrorToast, setShowErrorToast] = useState(false);
+const [showConfirmModal, setShowConfirmModal] = useState(false);
+const [proposals, setProposals] = useState([]);
+const [rowsPerPage, setRowsPerPage] = useState(10);
+const [currentPage, setPage] = useState(0);
+const [offset, setOffset] = useState(null);
+const [totalLength, setTotalLength] = useState(null);
+const [isPrevPageCalled, setIsPrevCalled] = useState(false);
+
+const fetchProposals = async () =>
+  getFilteredProposalsByStatusAndKind({
+    treasuryDaoID,
+    resPerPage: rowsPerPage,
+    isPrevPageCalled: isPrevPageCalled,
+    filterKindArray: ["ChangePolicy"],
+    filterStatusArray: ["InProgress"],
+    offset: typeof offset === "number" ? offset : lastProposalId,
+    lastProposalId: lastProposalId,
+    currentPage,
+  }).then((r) => {
+    setOffset(r.filteredProposals[r.filteredProposals.length - 1].id);
+    if (currentPage === 0 && !totalLength) {
+      setTotalLength(r.totalLength);
+    }
+
+    return r.filteredProposals;
```

Comment:
why do we need offset, currentpage and other such params, we just want to check if there is even a single pending proposal of `ChangePolicy`, can you remove the unnecessary params.